File: app/frontend/sentry.js
import * as Sentry from "@sentry/svelte";

export function sentry(config) {
  if (!config?.dsn) {
    return;
  }

  Sentry.init({
    dsn: config.dsn,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.replayIntegration(),
    ],
    tracesSampleRate: 1.0,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  });
}

File: server/config/initializers/sentry.rb
if Masks.mode.server?
  if dsn = Masks.conf.sentry_dsn
    Sentry.init do |config|
      config.dsn = dsn
      config.breadcrumbs_logger = %i[active_support_logger http_logger]
      config.traces_sample_rate = 1.0
      config.profiles_sample_rate = 1.0
      config.enabled_patches << :graphql
    end
  end
end
